esphome/components/lvgl/touchscreens.py

```diff
@@ -0,0 +1,46 @@
+import esphome.codegen as cg
+from esphome.components.touchscreen import CONF_TOUCHSCREEN_ID, Touchscreen
+import esphome.config_validation as cv
+from esphome.const import CONF_ID
+from esphome.core import CORE, TimePeriod
+
+from .defines import (
+    CONF_LONG_PRESS_REPEAT_TIME,
+    CONF_LONG_PRESS_TIME,
+    CONF_TOUCHSCREENS,
+)
+from .helpers import lvgl_components_required
+from .lv_validation import lv_milliseconds
+from .lvcode import lv, lv_add
+from .types import LVTouchListener
+
+PRESS_TIME = cv.All(lv_milliseconds, cv.Range(max=TimePeriod(milliseconds=65535)))
+CONF_TOUCHSCREEN = "touchscreen"
+TOUCHSCREENS_CONFIG = cv.maybe_simple_value(
+    {
+        cv.Required(CONF_TOUCHSCREEN_ID): cv.use_id(Touchscreen),
+        cv.Optional(CONF_LONG_PRESS_TIME, default="400ms"): PRESS_TIME,
+        cv.Optional(CONF_LONG_PRESS_REPEAT_TIME, default="100ms"): PRESS_TIME,
+        cv.GenerateID(): cv.declare_id(LVTouchListener),
+    },
+    key=CONF_TOUCHSCREEN_ID,
+)
+
+
+def touchscreen_schema(config):
+    value = cv.ensure_list(TOUCHSCREENS_CONFIG)(config)
+    if value or CONF_TOUCHSCREEN not in CORE.loaded_integrations:
+        return value
+    return [TOUCHSCREENS_CONFIG(config)]
+
+
+async def touchscreens_to_code(var, config):
+    for tconf in config.get(CONF_TOUCHSCREENS) or ():
+        lvgl_components_required.add(CONF_TOUCHSCREEN)
+        touchscreen = await cg.get_variable(tconf[CONF_TOUCHSCREEN_ID])
+        lpt = tconf[CONF_LONG_PRESS_TIME].total_milliseconds
+        lprt = tconf[CONF_LONG_PRESS_REPEAT_TIME].total_milliseconds
+        listener = cg.new_Pvariable(tconf[CONF_ID], lpt, lprt)
+        await cg.register_parented(listener, var)
+        lv.indev_drv_register(listener.get_drv())
+        lv_add(touchscreen.register_listener(listener))
```

Comment:
Tested this locally and it is working with the normal cg instead of being placed inside the init lambda
```suggestion
        cg.add(touchscreen.register_listener(listener))
```

Reply:
As you saw, I've done that, but it may have to be reverted in the future. There is a potential for touchscreen events to be sent to the listener before the the parent component has been completely setup. Right now I can't see that will cause any problems.